### ISSUE TITLE
Getting latest typings for Excel Add-In in Visual Studio Excel Web Add-In

### ISSUE THREAD

**Issue by helpyhelperton** [OfficeDev/office-js#855]
Have tried all sorts of different build environments in order to build an Excel Add-In, eventually settling on a Visual Studio 2019 Excel Web Add-In project. Spent more time trying to get tutorial Excel code running within the sample project template in TypeScript, eventually discovering that some properties and method calls used in Script Lab example were not being recognised by VS. Determined that this was because they were supported in higher versioned requirements sets than the requirements set typings supported in **office-js.TypeScript.DefinitelyTyped** on NuGet.

I have read [https://github.com/OfficeDev/office-js/issues/207](url). 

## Expected Behaviour
1. Ability to install up-to-date **office-js.TypeScript.DefinitelyTyped** NuGet package with typing supporting **ExcelApi 1.10**. Not holding my breath.

2. Workaround where I can get my hands on the raw up-to-date file and add it to my project manually.

3. Least attractive option would be trying to figure out how to add Node to my project and installing **@types/office-js** (although I've just done that in a separate folder and see that index.d.ts still doesn't specify required properties and methods).

## Current Behaviour
I'm -->this<-- close to having the project build after adding 

`<TypeScriptLib>DOM,ES2015</TypeScriptLib>`

in all the `<PropertyGroup>` nodes to both csproj files.

**Home.ts**
```
    async function playWithWorkbook() {
        await Excel.run(async (context) => {
            let xlWorkbook = context.workbook;
            xlWorkbook.load("name");
            await context.sync();
            console.log(`The workbook name is ${xlWorkbook.name}.`);
        });
    }
```
**name** is highlighted in the code window and reported as an error in the Error List.

`(TS) Property 'name' does not exist on type 'Workbook'. Did you mean 'names'?`

Don't get excited. I'm not that dumb (usually). 'names' is suggested because it is available in ExcelApi 1.1 and therefore present in **office-js.TypeScript.DefinitelyTyped**. 'name' is only available in ExcelApi 1.7 and therefore **not** available in said typings. I have the same problem with

`(TS) Property 'getCount' does not exist on type 'WorksheetCollection'.`

elsewhere in Home.ts.

## Steps to Reproduce
  1. Create new project and select Excel Web Add-In project template
  2. For add-in type choose **Add new functionalities to Excel**.
  3. Follow these steps: [Convert JavaScript to TypeScript](https://docs.microsoft.com/en-us/office/dev/add-ins/develop/convert-javascript-to-typescript)

## Context
I've been trying to put together a debugging environment that I'm comfortable with for developing the Excel Add-In I'm working on. I had the following criteria:
1. Debugging in an IDE not browsers.
2. Debugging in a fully fledged project not prototyping in Script Lab.
3. Preferably debug an Add-In for Desktop Excel to ensure the closest environment to what users use.

So I've tried the following:
1. Yeoman generated an Add-In project for Visual Studio Code.
   After a couple of days trying to get debugging to work, read this [here](https://code.visualstudio.com/docs/other/office).

> VS Code does not currently support client-side debugging. To debug your client-side Add-in, you can use the Office web clients and open the browser's developer tools and debug the Add-in just like any other client-side JavaScript application.

No way, José.

2. Various Visual Studio Excel Web Add-In template projects with different customisations.
I tried to replicate the Add-In from this [tutorial](https://www.youtube.com/watch?v=F-ORtU1IE_4). Source [here](https://gist.github.com/areed1192/ab867da7eb6292e0fcb81c98dee4cda9).

i. I migrated the Script Lab code from the tutorial into a normal JS Add-In project and immediately ran into syntax errors at runtime related to async/await. Read [this](https://docs.microsoft.com/en-us/office/dev/add-ins/develop/asynchronous-programming-in-office-add-ins).

> The JavaScript API for Office supports two kinds of asynchronous programming patterns:
>  * Using nested callbacks
>  * Using the promises pattern

Groan. Nested callbacks or marginally better **.then** ladders.

Did more research to educate my non-Web oriented self. Confusion heightened when it became clear that JavaScript supports async/await. But it was falling over whenever it saw an async. I've now developed a working theory that Desktop Excel is running a lower than ECMAScript 6 runtime. **If anyone can confirm this, it would help my understanding of how the architecture hangs together. Also, how does TypeScript run in Script Lab in Desktop Excel? Is Script Lab running a separate runtime?**

ii. Realised I would have to try and convert a JS project to TypeScript to get async/await and myriad other syntactical sugars working otherwise I'd be doomed to **.thens**. 

Proceeded to convert project to TypeScript using [this](https://docs.microsoft.com/en-us/office/dev/add-ins/develop/convert-javascript-to-typescript).

I also added  the following in all the `<PropertyGroup>` nodes in both csproj files to get it to recognise async/await and other later JS syntax.

`<TypeScriptLib>DOM,ES2015</TypeScriptLib>`

But once I had done that, the syntax issue disappeared. But I've been left with this typing issue.

If I replace those unrecognised properties or method calls, the Add-In launches in Excel and I can step through the code.

So, basically. I'm only missing up-to-date typing before I can continue on my merry way. I'll be pretty bummed out if I have to go back to a JS project which only supports ECMAScript 5 and promises when I'm this close.

## Your Environment

* Platform [PC desktop, Mac, iOS, Office Online]: PC desktop
* Host [Excel, Word, PowerPoint, etc.]: Excel
* Office version number: 1910 Build 12130.20344
* Operating System: Windows 7 Ultimate SP1

**Comment by sumurthy** [comment]
Tagging folks to see if this can be fixed. @weshi isn't @types/office-js updated with ExcelApi1.10?

**Comment by AlexJerabek** [comment]
@sumurthy [Definitely Typed](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/office-js/index.d.ts) is updated, but the nuget package is not. This is the problem we were discussing with @kbrandl.

**Comment by helpyhelperton** [comment]
Gents, before you spend anymore time on this, I abandoned this effort in the interests of time and have reverted to yeoman project with VS Code and debugging Excel Online add-in in the browser.

That said, @AlexJerabek explanation makes a lot of sense and I still have the project hanging around so if the NuGet package will be updated, I'm happy to spend some time to give it a run so that others may benefit from this issue in the future if they want to go the Visual Studio route. I've gone too far down the road with my current project for it to be of any use to me now, though. Let me know.

**Comment by AlexJerabek** [comment]
Pinging this thread to say that ExcelApi 1.10 in currently in the nuget file, but 1.11 and 1.12 are not. @sumurthy & @weshi, is there any way to get regular updates to the nuget file or have that update be part of our requirement set release cadence?
